### PR TITLE
[draft] doc: http keyword updates 3025 v2

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3025

Describe changes:
- updated http keyword documentation
- incorporated initial feedback from v1
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by jmtaylor90** [comment]
https://rtd-test-suricata.readthedocs.io/en/latest/rules/http-keywords.html for builds of this branch

**Comment by catenacyber** [comment]
Best PR number so far 👍

File: doc/userguide/rules/http-keywords.rst [hunk 1]
```diff
@@ -365,11 +410,9 @@ Normalization: leading spaces **are not** part of this buffer. So
 
 Example of the User-Agent header in a HTTP request:
 
-.. image:: http-keywords/user_agent.png
 
 Example of the purpose of ``http.user_agent``:
```
Comment (jufajardini) [inline doc/userguide/rules/http-keywords.rst:415]: Missing removal, or missing adding new examples?

Comment (jufajardini) [reply doc/userguide/rules/http-keywords.rst:415]: I've observed a few other cases like this, further down.

**Review by jufajardini — CHANGES_REQUESTED** [review]
Agreeing with Philippe on the PR number comment P:

Overall, looking good. 👏🏽 

Left a question related to the "Example of..." bits left after images' removal, I'm guessing those are placeholders for new examples to come?

One thing to note about the commit messages: some of them have abandoned the usage of imperative that we use. Sorry if this is annoying, but could you fix that?

Thanks a lot for all this work!!

File: doc/userguide/rules/http-keywords.rst [hunk 2]
```diff
@@ -556,12 +471,10 @@ like ``distance``, ``offset``, ``nocase``, ``within``, etc.
 
 Example of ``http.request_body`` in a HTTP request:
 
-.. image:: http-keywords/client_body.png
 
 Example of the purpose of ``http.client_body``:
 
-.. image:: http-keywords/client_body1.png
-
+1
```
[the remark this answers is not in the document]
Comment (jmtaylor90) [reply doc/userguide/rules/http-keywords.rst:621]: Yeah that will be fixed in next PR

File: doc/userguide/rules/http-keywords.rst (hunk 1 above)
Comment (jmtaylor90) [reply doc/userguide/rules/http-keywords.rst:415]: That's from the request to remove all the images in one commit. The images were removed and the artifacts left so when each keyword is reviewed an example is at least considered again.

**Comment by jmtaylor90** [comment]
> Agreeing with Philippe on the PR number comment P:
> 
> Overall, looking good. 👏🏽
> 
> Left a question related to the "Example of..." bits left after images' removal, I'm guessing those are placeholders for new examples to come?
> 
> One thing to note about the commit messages: some of them have abandoned the usage of imperative that we use. Sorry if this is annoying, but could you fix that?

Ah yeah, I will get those all amended for the next PR

> Thanks a lot for all this work!!

File: doc/userguide/rules/http-keywords.rst [hunk 3]
```diff
@@ -1,351 +1,396 @@
...
+  content:"User-Agent|3a 20|Mozilla/5.0|0d 0a|"; startswith;
+  content:"Host|3a 20|suricata.io|0d 0a|";` classtype:bad-unknown; \
+  sid:72; rev:1;)
+
+.. note:: There are headers that will not be included in the ``http.header`` keywords,
+  specifically the :ref:`http.cookie` buffer.
+
+.. note:: If there are multiple values for the same header name, they are concatenated
+  with a comma and space (", ") between each value. More information can be
+  found in RFC 2616 `<https://www.ietf.org/rfc/rfc2616.txt>`_ , section 4.2
```
[the remark this answers is not in the document]
Comment (jmtaylor90) [reply doc/userguide/rules/http-keywords.rst:335]: Meaning put the link below the paragraph?

File: doc/userguide/rules/http-keywords.rst (hunk 1 above)
Comment (jufajardini) [reply doc/userguide/rules/http-keywords.rst]: Fair, better than just removing them and end up forgetting. Thanks for taking this into consideration! :)

**Comment by catenacyber** [comment]
@jmtaylor90 I see you force-pushed, did you mean to open a new PR with changes ?

**Comment by jmtaylor90** [comment]
I think I have enough feedback to get this done. Thank you all.

**Pull request closed without merging**